### DISCUSSION TITLE
Cannot add "email" provider to an existing user

### DISCUSSION THREAD

**Discussion by ttskch** [supabase/supabase#12986] [Feature Requests]
### Case with no problems

1. User signs up with `email` provider.
2. A record is inserted to the `auth.users` table and `["email"]` is set in the `app_metadata.providers`.
3. A record is inserted to the `auth.identities` table.
4. User signs in with `signInWithOAuth()` using a Twitter account that has the same email as the one registered in step 1.
5 A record is inserted to the `auth.identities` table.
6. Since the emails match, the user is considered the same and the `app_metadata.providers` in `auth.users` is updated to `["email", "twitter"]`.
7. From now on, the user can sign in with either email or twitter.

### Case with problems

1. User signs up with a `twitter` provider.
2. A record is inserted to the `auth.users` table and `["twitter"]` is set in the `app_metadata.providers`.
3. A record is inserted to the `auth.identities` table.
4. User tries to `signUp()` with the same email as the Twitter account used in step 1.
5. `"User already registered."` error occurs.
6. There is no way to add a record to the `auth.identities` table corresponding to the `email` provider for this user.

**Comment by ioucyf** [comment]
It also throws an error when a user changes their email to an unrecorded address, then tries to change it back to one that is already associated to one of their identities; which makes it impossible without admin/postgres functions.
